ui-templates/README.md

```diff
@@ -0,0 +1,51 @@
+# UI Templates
+These JSON objects define templates for capturing user data from a UI and using it to set the parameters of SimulariumIO objects in order to visualize simulation trajectories.
+
+
+## Data types
+Different types of data need to be collected. Some of the data types are "base" HTML Form Input types, and some are "custom" types that contain parameters of the base types or other custom types. Some custom types are "main" types for visualizing data from a specific simulation engine.
+
+Each custom data type has:
+* `python::module` - Python module
+* `python::object` - Python class for the object to populate
+* `parameters` - a dict of names of members of the Python class mapped to information about how to set them. For each:
+  * `name` - the name to display next to the UI Input element
+  * `data_type` - the data type to collect, either a base or custom data type
+  * `description` - instructions for the user about how to fill this field
+  * `required` - is this field required in order to submit?
+  * `default` - default value to use for this field (only for user information)
+  * `validation` - a regex pattern to validate the data against (optional)
+  * `help` - additional information to help the user if the value doesn't pass validation (optional)
+  * some types also have additional parameter information
+
+### Main types
+* `smoldyn_data` - information for populating a `SmoldynData` to visualize a trajectory from Smoldyn
+* (more coming soon)
+
+### Base types
+
+* `number`
+* `text`
+* `url`
+* `color`
+
+### Custom types
+
+* `meta_data`
+* `coordinates_xyz`
+  * parameters are indices of items in a list to be provided as a single argument to `numpy.array()`
+* `camera_data`
+* `model_meta_data`
+* `collection` - collect an extensible list of information and use it to create a dict
+  * requires additional parameter information:
+    * `length` - number of items to start with
+    * `extendible`- can the user add more items?
+    * `key_item` - requires the same fields as `parameters` items
+    * `value_item` - requires the same fields as `parameters` items
+* `display_data`
```

Comment:
you call `display_data` a data type here, but in the `smoldyn_data`, display_data's `data_type` is set to `collection`

Reply:
`display_data` is the name of a parameter on the `SmoldynData` object that is a collection of `DisplayData`s for each agent type. It's confusing that it has the same name as a template type, but I don't think it makes sense to change either of them

Reply:
It's reasonable to disallow the reuse of data type names as data value names. But at the same time, we usually know where in the data we expect to see a type name vs a value name. In programming languages, usually you have a convention to name `DataType` with different casing than a `data_value`(Python) or `dataValue`(TS/JS/C++) of that type.  

Can you have `collection`s of different data types or is a collection always expected to be of the same type?  If always of a consistent type, you could replace `collection` with `collection[display_data]` or `display_data[]` shorthand notation that is easy enough to parse to see that it's intended to be a collection.  Just a thought but don't know if it's viable with this code.